packages/angular/build/src/tools/esbuild/angular/compiler-plugin.ts

```diff
@@ -588,11 +588,14 @@ export function createCompilerPlugin(
         logCumulativeDurations();
       });
 
-      build.onDispose(() => {
-        sharedTSCompilationState?.dispose();
-        void compilation.close?.();
-        void cacheStore?.close();
-      });
+      build.onDispose(
+        () =>
+          void Promise.all(
+            [compilation?.close?.(), cacheStore?.close(), javascriptTransformer.close()].filter(
```

Comment:
If you do not mind, I might create (or you can if you wish too), to actually call  include `void javascriptTransformer.close();` so that it's included in today's release.